modules/ting_das/ting_das.module

```diff
@@ -0,0 +1,378 @@
+
+    // Check that no one tries to order not supported type by mistake.
+    if (isset($types[$type])) {
+      $form = drupal_get_form('ting_das_order_form', $entity);
+      $content = drupal_render($form);
+      $commands[] = ajax_command_ding_popup('ting_das', t('Order digital copy'), $content, array('refresh' => TRUE));
+    }
+    else {
+      $commands[] = ajax_command_ding_popup('ting_das', t('Error'), '<p>' . t('Material type is not available for ordering.') . '</p>', array('refresh' => FALSE));
+    }
+  }
+
+  // Return the ajax commands as an render array.
+  return array('#type' => 'ajax', '#commands' => $commands);
+}
+
+/**
+ * Implements hook_form().
+ *
+ * Defines the place order form.
+ */
+function ting_das_order_form($form_id, &$form_state, TingEntity $entity) {
+  $form = array(
+    '#attributes' => array(
+      'class' => array(
+        'digital-order-form',
+      ),
+    ),
+  );
+
+  global $user;
+  $provider = _ding_provider_get_provider('user');
+  $profile = ding_user_provider_profile($user);
+  if ($profile) {
+    $profile = entity_metadata_wrapper('profile2', $profile);
+  }
+
+  $texts = variable_get('ting_das_text', array());
+  $creds = ding_user_get_creds();
+
+  if (isset($creds['resident']) && $creds['resident']) {
+    $form['title'] = array(
+      '#markup' => '<div class="popupbar-title"><strong>' . t('Fill out the form to order at digital copy') . '</strong></div>',
+    );
+
+    $form['description'] = array(
+      '#prefix' => '<div class="form-item form-type-markup description">',
+      '#markup' => check_markup($texts['description']['value'], $texts['description']['format']),
+      '#suffix' => '</div>',
+    );
+
+    $form['entity'] = array(
+      '#tree' => TRUE,
+    );
+
+    $form['entity']['pid'] = array(
+      '#type' => 'value',
+      '#value' => $entity->getId(),
+    );
+
+    $form['entity']['title'] = array(
+      '#type' => 'value',
+      '#value' => $entity->getTitle(),
+    );
+
+    $form['loaner_id'] = array(
+      '#type' => 'value',
+      '#value' => $creds['library_card_number'],
+    );
+
+    $form['name'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Name'),
+      '#default_value' => $profile ? $profile->{'field_' . $provider['module'] . '_name'}->value() : '',
+      '#required' => TRUE,
+    );
+
+    $form['mail'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Mail address'),
+      '#default_value' => $profile ? $profile->{'field_' . $provider['module'] . '_mail'}->value() : '',
+      '#element_validate' => array(
+        'ting_das_mail_validate',
+      ),
+      '#required' => TRUE,
+    );
+
+    $function = $provider['module'] . '_user_branches';
+    $preferred_branch = $profile ? $profile->{'field_' . $provider['module'] . '_preferred_branch'}->value() : '';
+    $form += ding_reservation_default_options_branch('fbs_preferred_branch', $preferred_branch, $function($user));
```

Comment:
It seems as if this creates a dependency on `fbs`? That may be OK but it is not declared.

Reply:
Added as dependency in info file now. It's required as KB wants the PID in the request sent to them. So as I see it the dependency is there.